### PR TITLE
Add child abuse report reason

### LINKED ISSUE #5827: Add option to report a message as Child Abuse
Android official clients have option to report a message as child abuse. Please add it in Tdesktop too.

### PR DESCRIPTION
**Pull request by omkarnathsingh**
Fixes #5827

### CODE REVIEW

**Comment by omkarnathsingh** [comment]
@john-preston Can you please have a look? Thanks,

**Comment by omkarnathsingh** [comment]
Because Preston said "Great! But this reason should appear only when you're reporting single messages, not when you're reporting a bot or a channel from their profile."

**Comment by omkarnathsingh** [comment]
@john-preston can you now review? Thanks.

File: Telegram/SourceFiles/boxes/report_box.cpp [hunk 1]
```diff
@@ -181,6 +191,11 @@ bool ReportBox::reportFail(const RPCError &error) {
 
 void ReportBox::updateMaxHeight() {
 	auto newHeight = st::boxOptionListPadding.top() + _reasonSpam->getMargins().top() + 4 * _reasonSpam->heightNoMargins() + 3 * st::boxOptionListSkip + _reasonSpam->getMargins().bottom() + st::boxOptionListPadding.bottom();
+
+	if (_ids) {
+		newHeight = st::boxOptionListPadding.top() + _reasonSpam->getMargins().top() + 5 * _reasonSpam->heightNoMargins() + 4 * st::boxOptionListSkip + _reasonSpam->getMargins().bottom() + st::boxOptionListPadding.bottom();
+	}
+
```
Comment (john-preston) [inline Telegram/SourceFiles/boxes/report_box.cpp]: I'd suggest something like:
```
const auto buttonsCount = _ids ? 5 : 4;
auto newHeight = st::boxOptionListPadding.top() + _reasonSpam->getMargins().top() + buttonsCount * _reasonSpam->heightNoMargins() + (buttonsCount - 1) * st::boxOptionListSkip + _reasonSpam->getMargins().bottom() + st::boxOptionListPadding.bottom();
```
without the new condition.

**Comment by john-preston** [comment]
Thanks!

**Pull request merged**